luigi/execution_summary.py

```diff
@@ -32,6 +33,66 @@ class execution_summary(luigi.Config):
+    the argument ``detailed_summary=True`` in *luigi.run() / luigi.build*.
+    Here are the codes and what they mean:
+
+    =============================  ==========================================================
+    Status Code Name               Meaning
+    =============================  ==========================================================
+    SUCCESS                        There were no failed tasks or missing dependencies
+    SUCCESS_WITH_RETRY             There were failed tasks but they all succeeded in a retry
+    FAILED                         There were failed tasks
+    FAILED_AND_SCHEDULING_FAILED   There were failed tasks and tasks whose scheduling failed
+    SCHEDULING_FAILED              There were tasks whose scheduling failed
+    NOT_RUN                        There were tasks that were not granted run permission by the scheduler
+    MISSING_EXT                    There were missing external dependencies
+    =============================  ==========================================================
+
+    """
+    SUCCESS = (":)", "there were no failed tasks or missing dependencies")
+    SUCCESS_WITH_RETRY = (":)", "there were failed tasks but they all succeeded in a retry")
+    FAILED = (":(", "there were failed tasks")
+    FAILED_AND_SCHEDULING_FAILED = (":(", "there were failed tasks and tasks whose scheduling failed")
+    SCHEDULING_FAILED = (":(", "there were tasks whose scheduling failed")
+    NOT_RUN = (":|", "there were tasks that were not granted run permission by the scheduler")
+    MISSING_EXT = (":|", "there were missing external dependencies")
+
+
+class LuigiRunResult(object):
+    """
+    The result of a call to build/run when passing the detailed_summary=True argument.
+
+    Attributes:
+        - summary_text_one_line (str): One line summary of the progress.
+        - summary_text (str): Detailed summary of the progress.
+        - status (LuigiStatusCode): Luigi Status Code. See :class:`~luigi.execution_summary.LuigiStatusCode` for what these codes mean.
+        - worker (luigi.worker.worker): Worker object. See :class:`~luigi.worker.worker`.
+        - execution_succeeded (bool): Boolean which is *True* if finally, there were no failed tasks.
+        - scheduling_succeeded (bool): Boolean which is *True* if all the tasks were scheduled without errors.
+
+    """
+    def __init__(self, worker, worker_add_run_status=True):
+        self.worker = worker
+        summary_dict = _summary_dict(worker)
+        self.summary_text = _summary_wrap(_summary_format(summary_dict, worker))
+        self.status = _tasks_status(summary_dict)
+        self.summary_text_one_line = _progress_summary(self.status)
+        self.scheduling_succeeded = worker_add_run_status
+
+    # This function makes this class subscriptable (self.worker is used at a few places)
+    def __getitem__(self, key):
+        return getattr(self, key, None)
```

Comment:
I don't understand your comments, why is this necessary??

Reply:
@Tarrasch This function makes this class **subscriptable like a dictionary** for backwards compatibility.
- Previously a dictionary was returned by `_schedule_and_run` (in `interface.py`) which had the structure: `{ "success": True, "worker": "<worker>"}`. 
- Now, a `luigi_run_result` (object of type `LuigiRunResult`) is returned which doesn't have the key `luigi_run_result["worker"]` for a few existing functions to use. It is currently an attribute (`luigi_run_result.worker`).
- This function makes the class subscriptable for _dictionary_ like interaction.

Reply:
Actually some tests were failing because of this.
So should I change those internal `["worker"]` accesses to `.worker` as well then ?

Reply:
There was just one reference in `retcodes.py`. Made the change 👍